xicam/gui/bluesky/search.py

```diff
@@ -176,14 +183,16 @@ def set_selected_catalog(self, item):
         self.selected_catalog = self.catalog[name]()
         self.search()
 
+    @timeit
     def check_for_new_entries(self):
         # check for any new results and add them to the queue for later processing
-        for uid, entry in itertools.islice(self._results_catalog.items(), MAX_SEARCH_RESULTS):
-            if uid in self._results:
-                continue
-            self._results.append(uid)
-            self._new_entries.put(entry)
+        for uid in list(self._results_catalog)[:MAX_SEARCH_RESULTS]:
```

Comment:
This will make the performance much worse for large results sets because it will pull the entire results set into memory, rather than just the first `MAX_SEARCH_RESULTS` entries. I don't think we can do this.

Reply:
Thinking on this some, I'm completely stumped as to where that speedup would come from. In this toy example, you can see that `list(...)[:N]` accesses _every_ element, including the ones past `N` that aren't needed, whereas `itertools.islice(..., N)` accesses just the elements we need.

```py
In [8]: def my_iterator(): 
   ...:     for i in range(10): 
   ...:         print(f'accessing item {i}') 
   ...:         yield i 
   ...:                                                                                                                                                                                       

In [9]: for _ in list(my_iterator())[:5]: 
   ...:     ... 
   ...:                                                                                                                                                                                       
accessing item 0
accessing item 1
accessing item 2
accessing item 3
accessing item 4
accessing item 5
accessing item 6
accessing item 7
accessing item 8
accessing item 9

In [10]: import itertools                                                                                                                                                                     

In [11]: for _ in itertools.islice(my_iterator(), 5): 
    ...:     ... 
    ...:                                                                                                                                                                                      
accessing item 0
accessing item 1
accessing item 2
accessing item 3
accessing item 4
```